app/src/lib/hostedServices/interface/hostedGitPrMonitor.ts

```diff
@@ -0,0 +1,17 @@
+import { buildContextStore } from '$lib/utils/context';
+import type { DetailedPullRequest, MergeMethod } from './types';
+import type { Readable } from 'svelte/store';
+
+export const PR_MONITOR = Symbol('PullRequestMonitor');
+
+export interface HostedGitPrMonitor {
+	pr: Readable<DetailedPullRequest | undefined>;
+	loading?: Readable<boolean>;
+	lastFetch?: Readable<Date | undefined>;
+	merge(method: MergeMethod): Promise<void>;
+	refresh(): Promise<void>;
+}
+
+export const [getHostedGitPrMonitorStore, createHostedGitPrMonitorStore] = buildContextStore<
+	HostedGitPrMonitor | undefined
+>('prMonitor');
```

Comment:
Because `HostedGitPrMonitor` is an interface and our current `setContext` works only with classes.

Reply:
So, it looks like we're using derived to create the instances, so a store makes sense; but if we weren't or could avoid that, it would be nice to have a buildContext which didn't make use of a store